src/compiler/checker.ts

```diff
@@ -936,6 +937,25 @@ namespace ts {
             return false;
         }
 
+
+        function checkAndReportErrorForExtendingInterface(errorLocation: Node, name: string): boolean {
+            if (!errorLocation || errorLocation.kind !== SyntaxKind.Identifier ||
+                !errorLocation.parent || !errorLocation.parent.parent ||

```

Comment:
The intention here is that if one wrongly extends interface, then the errorLocation must be exactly at `HeritageClause -> ExpressionWithTypeArguments -> Identifier`.